File: Tarea/app/src/main/java/com/example/practica01/MainActivity.java
package com.example.practica01;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button btnMultRusa;
    Button btnPalindrome;
    Button btnBisiesto;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnMultRusa=(Button) findViewById((R.id.btnMultRusa));
        btnPalindrome=(Button) findViewById((R.id.btnPalindrome));
        btnBisiesto=(Button) findViewById((R.id.btnBisiesto));

        btnMultRusa.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getBaseContext(),activity_MultiplicacionRusa.class);
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                getBaseContext().startActivity(intent);
            }
        });
        btnPalindrome.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getBaseContext(),activity_Palindrome.class);
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                getBaseContext().startActivity(intent);
            }
        });
        btnBisiesto.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getBaseContext(),activity_Bisiesto.class);
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                getBaseContext().startActivity(intent);
            }
        });
    }
}

File: SharedPreferencesAristiDevsExample2/app/src/main/java/com/example/sharedpreferencesaristidevsexample2/MainActivity.kt
package com.example.sharedpreferencesaristidevsexample2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText
import android.widget.TextView
import com.example.sharedpreferencesaristidevsexample2.UserVipApplication.Companion.prefss

class MainActivity : AppCompatActivity() {
    private lateinit var btnContinue: Button
    private lateinit var etName: EditText
    private lateinit var cbVip: CheckBox

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnContinue = findViewById(R.id.btnContinue)
        etName = findViewById(R.id.etName)
        cbVip = findViewById(R.id.cbVip)
        initUI()
        checkUserValues()
    }

    private fun checkUserValues() {
        if (prefss.getName().isNotEmpty()) {
            goToDetail()
        }
    }

    private fun initUI() {
        btnContinue.setOnClickListener { accessToDetail() }
    }

    private fun accessToDetail() {
        if (etName.text.toString().isNotEmpty()) {
            // Guardar usuario
            prefss.saveName(etName.text.toString())
            prefss.saveVIP(cbVip.isChecked)
            goToDetail()
        } else {
        }
    }

    fun goToDetail() {
        startActivity(Intent(this, ResultActivity::class.java))
    }
}